### PR TITLE
API-24715/ DocServer: Update Kong gateway logic to autoresolve service ids

### PR DESCRIPTION
**Pull request by mheusgit**
- API version create/update logic now checks against gateway to find if a route name matches to a service name.
- serviceIds are no longer allowed when creating/updating a version.  Gateway now controls this.

### CODE REVIEW

**Comment by mheusgit** [comment]
This PR is a good example why tickets such as API-23423 are needed.  3/4 of changes line changes are just for unit tests.  Same is true for 5 of every 6 files updated.

File: src/gateway/gateway.service.ts [hunk 1]
```diff
@@ -116,6 +119,60 @@ class GatewayService {
...
+    }
+
+    return resolvedConfig;
+  }
+
+  private async resolveServiceNameByRoute(routeName: string): Promise<string> {
+    try {
+      logger.info(`Fetching service ID for route '${routeName}'...`);
+      const routeConfigURL = `/routes/${routeName}`;
+      const kongRoute = await this.client.get<KongServiceRoute>(routeConfigURL);
```
[the remark this answers is not in the document]
Comment (mheusgit) [reply src/gateway/gateway.service.ts:141]: I considered it.  Once the need for serviceId being passed is removed (comment above) its better just to update the method signature to pass a string and then this method could call it without creating a phony gateway config object.

File: src/gateway/gateway.service.ts [hunk 2]
```diff
@@ -116,6 +119,60 @@ class GatewayService {
...
+      const serviceId = kongRoute.data.service.id;
+
+      if (!serviceId) {
+        throw Error(`Service ID was not found in response`);
+      }
+
+      logger.info(`Fetching service name for ID '${serviceId}'...`);
+      const serviceConfigURL = `/services/${serviceId}`;
+      const kongService = await this.client.get<KongService>(serviceConfigURL);
```
[the remark this answers is not in the document]
Comment (mheusgit) [reply src/gateway/gateway.service.ts:150]: Its a possibility but I'd want to have the invalid serviceName check and want the more detailed error logging.

File: src/gateway/gateway.service.ts [hunk 3]
```diff
@@ -67,6 +68,8 @@ class GatewayService {
     const serviceId = gatewayConfig.serviceId;
     const routeId = gatewayConfig.routeId;
     try {
+      // Method is due for cleanup.  Its currently not possible to have a serviceId without a matching routeId in DocServer
+      //  Logic after 'if' statement appears to be unused
```
[the remark this answers is not in the document]
Comment (christineyost) [reply src/gateway/gateway.service.ts:72]: No, I think this code is quite old

File: src/gateway/gateway.service.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (christineyost) [reply src/gateway/gateway.service.ts:141]: I think that can be done now. The fetchKongRoute method hasn't been updated for 2 or 3 years - I think the potential cleanup was just missed

[the remark this answers is not in the document]
Comment (mheusgit) [reply src/gateway/gateway.service.ts]: Change in place

File: src/gateway/gateway.service.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (mheusgit) [reply src/gateway/gateway.service.ts]: Change in place

**Comment by mheusgit** [comment]
PR should be good to go

**Review by christineyost — APPROVED** [review]
Looks good!!

**Pull request merged**